app/controllers/concerns/application_multifactor_methods.rb

```diff
@@ -4,6 +4,7 @@ module ApplicationMultifactorMethods
   included do
     def redirect_to_new_mfa
       message = t("multifactor_auths.setup_required_html")
+      session["mfa_redirect_uri"] = request.path_info
```

Comment:
Could you add to the current redirect to mfa tests to test that `session["mfa_redirect_uri"]` is being set to the expected paths?